src/cpp/rtps/transport/shared_mem/SharedMemTransport.cpp

```diff
@@ -421,7 +449,7 @@ bool SharedMemTransport::send(
 
         // Segment overflow with discard policy doesn't return error.
         if (!shared_buffer &&
-                configuration_.segment_overflow_policy() == SharedMemTransportDescriptor::OverflowPolicy::DISCARD)
+                shm_default_overflow_policy == SharedMemTransportDescriptor::OverflowPolicy::DISCARD)
```

Comment:
This condition is always true now. Shall we remove it?

Reply:
Overflow policy removed, but the if remains because has different treatment overflow than exception in send, although we could return always true... or false, upper layer really doesn't care about the return.

Reply:
> the if remains because ...

Of course, I was just referring to the second condition (`shm_default_overflow_policy == SharedMemTransportDescriptor::OverflowPolicy::DISCARD`)